statics/workflows/flow-validation.yaml

```diff
@@ -0,0 +1,116 @@
+---
+UUID: "1c79a9d2-6fcb-4e10-7d40-78819b07781c"
+Name: "FlowValidation"
+Description: "Check Flow-Forwarding form Checkpoint-1 to Checkpoint-2"
+Parameters:
+  - Name: Checkpoint-1
+    Description: Select Checkpoint-1
+    Type: node
+  - Name: Checkpoint-2
+    Description: Select Checkpoint-2 (Optional)
+    Type: node
+  - Name: All
+    Description: Create Capture on all nodes between Checkpoint-1 and Checkpoint-2 (Select Only with Checkpoint-2)
+    Type: boolean
+  - Name: Protocol
+    Description: Select Protocol to create BPF Filter
+    Type: choice
+    Default: icmp
+    Values:
+      - Description: "ICMP"
+        Value: icmp
+      - Description: "TCP"
+        Value: tcp
+      - Description: "UDP"
+        Value: udp
+  - Name: Source-IP
+    Description: Enter Source IP for BPF Filter (Optional)
+    Type: string
+  - Name: Destination-IP
+    Description: Enter Destination IP for BPF Filter (Optional)
+    Type: string
+  - Name: ID/Port
+    Description: Enter The Destination Port or The ICMP-ID for BPF Filter (Optional)
+    Type: string
+  - Name: Time
+    Description: Time Duration to capture the flows in seconds (Optional - Default 5 seconds)
+    Type: integer
+Source: |
+  function FlowValidation(checkPoint1, checkPoint2, all, protocol, srcIP, dstIP, id, interval) {
+    try {
+      var capture = new Capture();
+      capture.SamplingRate = 1;
+      capture.PollingInterval = 10;
+
+      if (checkPoint2 == null) {
+        capture.GremlinQuery = "G.V().Has('TID', '" + checkPoint1 + "')";
+      } else if (checkPoint2 != null && !all) {
+        capture.GremlinQuery = "G.V().Has('TID', Within('" + checkPoint1 + "', '" + checkPoint2 + "'))";
+      } else {
+        capture.GremlinQuery = "G.V().Has('TID', '" + checkPoint1 + "').ShortestPathTo(Metadata('TID', '" + checkPoint2 + "'), Metadata('RelationType', 'layer2'))";
+      }
+
+      capture.BPFFilter = protocol;
+
+      if (srcIP && srcIP !== "") {
+        capture.BPFFilter += " and (src "+ srcIP + " or dst " + srcIP + ")";
+      }
+
+      if (dstIP && dstIP !== "") {
+        capture.BPFFilter += " and (src "+ dstIP + " or dst " + dstIP + ")";
+      }
+
+      if (protocol == "icmp" && id && id !== "") {
+        capture.BPFFilter +=  " and icmp[4:2]=" + id;
+      } else if (id && id !== "") {
+        capture.BPFFilter +=  " and port " + id;
+      }
+
+      capture = client.captures.create(capture)
+
+      if (interval && interval > 0) {
+        var timediff = interval * 1000;
+      }else {
+        var timediff = 5 * 1000;
+      }
+
+      sleep(timediff)
+      var flowExpected = client.gremlin.G().V().Has('Capture.ID', capture.UUID).result();
+      var flowCaptured = client.gremlin.G().V().Has('Capture.ID', capture.UUID).Flows().CaptureNode().result();
```

Comment:
use ```dedup``` step to remove the duplicate nodes.

Reply:
```client.gremlin.G().V().Has('Capture.ID', capture.UUID).Flows().CaptureNode().Dedup().result();```

Reply:
Done!